src/todos/todos.database.service.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { Test, TestingModule } from '@nestjs/testing'
+import { getModelToken } from '@nestjs/mongoose'
+import { Types } from 'mongoose'
+
+import { TodosDatabaseService } from './todos.database.service'
+import { NotFoundError } from 'src/common/errors/errors'
+import { TODO_MODEL } from 'src/common/constants/database.constants'
+import { UpdateTodoDto } from './dto/update-todo.dto'
+
+describe('TodosDatabaseService', () => {
+  let todosDatabaseService: TodosDatabaseService
+  const mockTodoModel = {
+    findByIdAndUpdate: jest.fn(),
+  }
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        TodosDatabaseService,
+        {
+          provide: getModelToken(TODO_MODEL),
+          useValue: mockTodoModel,
+        },
+      ],
+    }).compile()
+
+    todosDatabaseService =
+      module.get<TodosDatabaseService>(TodosDatabaseService)
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe('update()', () => {
+    const id = new Types.ObjectId().toString()
+    const updateParams: UpdateTodoDto = {
+      title: 'Updated title',
+      description: 'Updated description',
+    }
+    const updatedTodo = {
+      _id: id,
+      title: updateParams.title,
+      description: updateParams.description,
+      toObject: jest.fn(),
```

Comment:
Let's add some return value, otherwise in line 61 we're comparing `expect(undefined).toEqual(undefined)`; which means — this test will still pass even if you remove all the logic from the method being tested.